app/src/main/java/com/islaharper/dawnofandroid/presentation/screens/splash/SplashViewModel.kt

```diff
@@ -1,5 +1,25 @@
 package com.islaharper.dawnofandroid.presentation.screens.splash
 
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.islaharper.dawnofandroid.domain.useCases.UseCases
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import javax.inject.Inject
 
-class SplashViewModel : ViewModel()
+@HiltViewModel
+class SplashViewModel @Inject constructor(private val useCases: UseCases) : ViewModel() {
+    private val _onBoardingComplete = MutableStateFlow(false)
+    val onBoardingCompleted: StateFlow<Boolean> = _onBoardingComplete
+
+    init {
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
ViewModels shouldn't normally have to choose the Dispatcher. This is because the concept of main safety in coroutines. Can you please do some research about this and then refactor?

Reply:
Specifying the Dispatcher is an old habit, I'll make the change to use Default